crates/sncast/src/helpers/configuration.rs

```diff
@@ -1,7 +1,154 @@
 use std::fs;
+use tempfile::TempDir;
+use toml::Value;
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
+pub struct CastConfig {
+    pub rpc_url: String,
+    pub account: String,
+    pub accounts_file: Utf8PathBuf,
+    pub keystore: Option<Utf8PathBuf>,
+    pub wait_timeout: u16,
+    pub wait_retry_interval: u8,
+}
+
+impl CastConfig {
+    pub fn from_sncast_config(
+        sncast_config: &Value,
+        profile: &Option<String>,
+    ) -> Result<CastConfig> {
+        let entries = get_profile(sncast_config, profile)?;
+
+        Ok(CastConfig {
+            rpc_url: get_property(&entries, "url"),
+            account: get_property(&entries, "account"),
+            accounts_file: get_property(&entries, "accounts-file"),
+            keystore: get_property_optional(&entries, "keystore"),
+            wait_timeout: get_property(&entries, "wait-timeout"),
+            wait_retry_interval: get_property(&entries, "wait-retry-interval"),
+        })
+    }
+}
+
+impl Default for CastConfig {
+    fn default() -> Self {
+        Self {
+            rpc_url: String::default(),
+            account: String::default(),
+            accounts_file: Utf8PathBuf::default(),
+            keystore: None,
+            wait_timeout: WAIT_TIMEOUT,
+            wait_retry_interval: WAIT_RETRY_INTERVAL,
+        }
+    }
+}
+
+pub trait PropertyFromCastConfig: Sized {
+    fn from_toml_value(value: &Value) -> Option<Self>;
+    fn default_value() -> Self;
+}
+
+impl PropertyFromCastConfig for String {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_str().map(std::borrow::ToOwned::to_owned)
+    }
+
+    fn default_value() -> Self {
+        String::default()
+    }
+}
+
+impl PropertyFromCastConfig for Utf8PathBuf {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_str().map(Utf8PathBuf::from)
+    }
+
+    fn default_value() -> Self {
+        Utf8PathBuf::default()
+    }
+}
+
+impl PropertyFromCastConfig for u8 {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_integer().and_then(|i| i.try_into().ok())
+    }
+
+    fn default_value() -> Self {
+        WAIT_RETRY_INTERVAL
+    }
+}
+
+impl PropertyFromCastConfig for u16 {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_integer().and_then(|i| i.try_into().ok())
+    }
+
+    fn default_value() -> Self {
+        WAIT_TIMEOUT
+    }
+}
+
+impl<T> PropertyFromCastConfig for Option<T>
+where
+    T: PropertyFromCastConfig,
+{
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        T::from_toml_value(value).map(Some)
+    }
+    fn default_value() -> Self {
+        Some(T::default_value())
+    }
+}
+
+pub fn get_profile(sncast_config: &Value, profile: &Option<String>) -> Result<Value> {
+    let profile_name = profile.as_deref().unwrap_or("default");
+    let config = sncast_config
+        .get("sncast")
+        .expect("Failed to find sncast config in foundry.toml file");
```

Comment:
nit: foundry.toml -> snfoundry.toml

Reply:
Done in https://github.com/foundry-rs/starknet-foundry/pull/1644/commits/de9f39d88ada985c9bcb12ea3676aeee67b85684